.circleci/config.yml

```diff
@@ -25,14 +33,31 @@ jobs:
           paths:
             - ./venv
           key: v1-dependencies-{{ checksum "requirements.txt" }}
-        
+
+      - run:
+          name: remove ed formated private key
+          command: rm ~/.ssh/id_rsa
+
+      - run:
+          name: remove ed formated public key 
```

Comment:
This is a case where the auto generated SSH keys on the VM out of the box are ed25519 keys. My understanding is that  id_rsa and id_rsa.pub are both SSH keys in the ed25519 format. Hence, these two commands are deleting keys in the ed25519 format so that we can generate new keys without having to manually input that we want to overwrite existing keys. 

Happy to see if I can squash it all into a single commit. Will be a first time doing this, but I don't imagine it will be difficult to figure out.

Reply:
No id_rsa are not ed25519 format. https://linuxnatives.net/2019/how-to-create-good-ssh-keys

Reply:
Then I am not sure why I need to delete the old keys and then re generate them specifically with the rsa tag. Without this step I can't authenticate to azure.

Reply:
This is the error - 
![image](https://user-images.githubusercontent.com/91570951/178374956-19dc61ca-cd34-4dbc-bae2-1b7750ddce40.png)

Upon research one of the suggested problems was that Azure does not support ED25519 keys and thus authentication was failing. 

This may not be the issue, but something was wrong with the SSH keys that were on the container initially, when I deleted and re generated them all of my issues went away

Reply:
It looks like the original keys are created by Circle CI in the checkout process:

```
rm -f "$SSH_CONFIG_DIR/id_rsa"
printf "%s" "$CHECKOUT_KEY" > "$SSH_CONFIG_DIR/id_rsa"
chmod 0600 "$SSH_CONFIG_DIR/id_rsa"
if (: "${CHECKOUT_KEY_PUBLIC?}") 2>/dev/null; then
  rm -f "$SSH_CONFIG_DIR/id_rsa.pub"
  printf "%s" "$CHECKOUT_KEY_PUBLIC" > "$SSH_CONFIG_DIR/id_rsa.pub"
fi
```

I'm not sure where these keys come from or what format they are in. It would be interesting to find out. Clearly the keys are in a valid format or they wouldn't work in the code checkout. So why is the az command failing? As you say one reason is ed25519, but based on file name that shouldn't be the case.

Anyway, I think we are good to go here other than a misleading comment. probably just change that to something like "removing keys that are invalid for `az aks create` so that those who come later don't need to disect it.